runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -339,15 +345,20 @@ public final class MqttServerFactory implements MqttStreamFactory
 
     {
         final Map<MqttPacketType, MqttServerDecoder> decodersByPacketType = new EnumMap<>(MqttPacketType.class);
-        decodersByPacketType.put(MqttPacketType.CONNECT, decodeConnect);
-        decodersByPacketType.put(MqttPacketType.PUBLISH, decodePublish);
+        decodersByPacketType.put(MqttPacketType.CONNECTV4, decodeConnectV4);
+        decodersByPacketType.put(MqttPacketType.CONNECTV5, decodeConnectV5);
+        decodersByPacketType.put(MqttPacketType.PUBLISHV4, decodePublishV4);
+        decodersByPacketType.put(MqttPacketType.PUBLISHV5, decodePublishV5);
         // decodersByPacketType.put(MqttPacketType.PUBREC, decodePubrec);
         // decodersByPacketType.put(MqttPacketType.PUBREL, decodePubrel);
         // decodersByPacketType.put(MqttPacketType.PUBCOMP, decodePubcomp);
-        decodersByPacketType.put(MqttPacketType.SUBSCRIBE, decodeSubscribe);
-        decodersByPacketType.put(MqttPacketType.UNSUBSCRIBE, decodeUnsubscribe);
+        decodersByPacketType.put(MqttPacketType.SUBSCRIBEV4, decodeSubscribeV4);
+        decodersByPacketType.put(MqttPacketType.SUBSCRIBEV5, decodeSubscribeV5);
+        decodersByPacketType.put(MqttPacketType.UNSUBSCRIBEV4, decodeUnsubscribeV4);
+        decodersByPacketType.put(MqttPacketType.UNSUBSCRIBEV5, decodeUnsubscribeV5);
         decodersByPacketType.put(MqttPacketType.PINGREQ, decodePingreq);
-        decodersByPacketType.put(MqttPacketType.DISCONNECT, decodeDisconnect);
+        decodersByPacketType.put(MqttPacketType.DISCONNECTV4, decodeDisconnectV4);
+        decodersByPacketType.put(MqttPacketType.DISCONNECTV5, decodeDisconnectV5);
         // decodersByPacketType.put(MqttPacketType.AUTH, decodeAuth);
         this.decodersByPacketType = decodersByPacketType;
     }
```

Comment:
We need separate `decodersByPacketTypeV4` and `decodersByPacketTypeV5`, and no versions in the `MqttPacketType` constants.